presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroupManager.java

```diff
@@ -112,6 +112,7 @@ public final class InternalResourceGroupManager<C>
     private final boolean isResourceManagerEnabled;
     private final QueryManagerConfig queryManagerConfig;
     private final InternalNodeManager nodeManager;
+    private boolean isConfigurationManagerLoaded;
```

Comment:
Given that this is mutable and guaranteed to be accessed from multiple threads, you should ensure that there is proper visibility to each thread.  Note that where you set this, in `loadConfigurationManager`, such care has also been taken in other mutated members, like `configurationManager`.  I'd recommend making it an `AtomicBoolean`.

Reply:
will send a followup PR to address this.

Reply:
https://github.com/prestodb/presto/pull/23594